akka-actor/src/main/scala/akka/pattern/RetrySettings.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import akka.pattern.RetrySupport.calculateExponentialBackoffDelay
+
+import java.time.Duration
+import java.util.Optional
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.DurationConverters.JavaDurationOps
+
+final case class RetrySettings(
+    attempts: Int,
+    delayFunction: Int => Option[FiniteDuration],
+    shouldRetry: Throwable => Boolean = _ => true) {
+
+  override def toString: String = {
+    s"RetrySettings(attempts=$attempts, delayFunction=$delayFunction)"
+  }
+
+  def withDelay(delayFunction: Int => Option[FiniteDuration]): RetrySettings = {
```

Comment:
I think we should expose exponential backoff as a separate and more direct:
```
def withExponentialBackoff(minBackoff: FiniteDuration, maxBackoff: FiniteDuration, randomFactor: Double): RetrySettings
```

Then make `ExponentialBackoffFunction` internal.

Reply:
added, but keep in mind that we have a sort of creation dualism here:
```
RetrySettings(attempts, minBackoff, maxBackoff, randomFactor)
```
or 
```
RetrySettings(attempts). withExponentialBackoff(minBackoff, maxBackoff, randomFactor)
```
maybe that's perfectly fine, but just wanted to align with the existing Akka conventions.